### PR TITLE
Add simple pull examples with auth

### LINKED ISSUE #3569: Add Example of Pushing or Pulling from Registry w/ Credentials
File: [engine/api/get-started.md](https://docs.docker.com/engine/api/get-started/), CC @mstanleyjones

### PR DESCRIPTION
**Pull request by mstanleyjones**
### Proposed changes

Add simple pull-with-auth example

Fixes #3569

I'm really not sure about the Curl example.

cc/ @thaJeztah @alexmavr 

cc/ @shin- @dnephin for the Python example

### CODE REVIEW

File: engine/api/get-started.md [hunk 1]
```diff
@@ -586,6 +586,8 @@ func main() {
 		panic(err)
 	}
 
+  defer out.Close()
```
Comment (thaJeztah) [inline engine/api/get-started.md]: This is using spaces (other parts use tabs)

File: engine/api/get-started.md [hunk 2]
```diff
@@ -606,6 +608,101 @@ $ curl --unix-socket /var/run/docker.sock \
...
+<div id="tab-pullimages-auth-go" class="tab-pane fade" markdown="1">
+
+```go
+package main
+
+import (
+	"io"
+	"os"
+  "encoding/json"
+  "encoding/base64"
```
Comment (thaJeztah) [inline engine/api/get-started.md]: Mixed tabs/spaces here as well (see some other parts below that also mix tabs/spaces)

File: engine/api/get-started.md [hunk 3]
```diff
@@ -606,6 +608,101 @@ $ curl --unix-socket /var/run/docker.sock \
...
+      Username: "username",
+      Password: "password",
+  }
+  encodedJSON, err := json.Marshal(authConfig)
+  if err != nil {
+      return err
+  }
+  authStr := base64.URLEncoding.EncodeToString(encodedJSON)
+
+	out, err := cli.ImagePull(ctx, "alpine", types.ImagePullOptions{RegistryAUth:
```
Comment (thaJeztah) [inline engine/api/get-started.md]: `s/RegistryAUth/RegistryAuth/`

File: engine/api/get-started.md [hunk 4]
```diff
@@ -606,6 +608,101 @@ $ curl --unix-socket /var/run/docker.sock \
...
+
+<div class="tab-content">
+<div id="tab-pullimages-auth-python" class="tab-pane fade in active" markdown="1">
+
+```python
+import docker
+client = docker.from_env()
+image = client.images.pull("alpine", auth_config=['username','password'])
+print image.id
+```
```
Comment (shin-) [inline engine/api/get-started.md]: Hmm, the Python SDK retrieves auth info from the `~/.docker/config.json` file (and has integration with [credential helpers](https://github.com/docker/docker-credential-helpers)).

If for some reason you want to use a different set of credentials for this session, the easiest way to do so is as follows:
```python
import docker

client = docker.from_env()
client.login('username', 'password')
image = client.images.pull('alpine')
print(image.id)
```
The credentials provided in the call to `client.login` will be used for the rest of the session.

If one wants to use a specific credentials for a single pull, then `auth_config` parameter should be used as follows:

```python
import docker

client = docker.from_env()
image = client.images.pull('alpine', auth_config={"Username": "username", "Password": "password"})
print(image.id)
```

But honestly that second example is more advanced use (it bypasses a lot of checks and functionalities of the library) and probably doesn't belong on an introductory page.

Comment (mstanleyjones) [inline engine/api/get-started.md]: Does the Go SDK also integrate the same way, @shin- ?

Comment (shin-) [inline engine/api/get-started.md]: Sorry, I don't know about that.

**Comment by mstanleyjones** [comment]
@thaJeztah @shin- PTAL again

**Comment by shin-** [comment]
Python section LGTM

File: engine/api/get-started.md [hunk 5]
```diff
@@ -606,6 +608,109 @@ $ curl --unix-socket /var/run/docker.sock \
...
+    panic(err)
+  }
+
+  authConfig := types.AuthConfig{
+      Username: "username",
+      Password: "password",
+  }
+  encodedJSON, err := json.Marshal(authConfig)
+  if err != nil {
+      return err
```
Comment (alexmavr) [inline engine/api/get-started.md]: The `main` function does not return an error. This should be `panic(err)` instead

Comment (mstanleyjones) [inline engine/api/get-started.md]: Fixed

**Comment by mstanleyjones** [comment]
@thaJeztah just waiting for your LGTM

**Comment by mstanleyjones** [comment]
Merging 😱

**Pull request merged**